public/app/features/alerting/unified/hooks/useProduceNewRuleGroup.tsx

```diff
@@ -20,39 +29,19 @@ type ProduceResult = RulerRuleGroupDTO | AlertGroupUpdated;
  * and to guard against user concurrency issues.
  *
  * @throws
- * @TODO the manual state tracking here is not great, but I don't have a better idea that works /shrug
  */
 function useProduceNewRuleGroup() {
-  const [fetchRuleGroup] = alertRuleApi.endpoints.getRuleGroupForNamespace.useLazyQuery();
-  const [updateRuleGroup] = alertRuleApi.endpoints.updateRuleGroupForNamespace.useMutation();
-  const [deleteRuleGroup] = alertRuleApi.endpoints.deleteRuleGroupFromNamespace.useMutation();
-
-  const [isLoading, setLoading] = useState<boolean>(false);
-  const [isUninitialized, setUninitialized] = useState<boolean>(true);
-  const [result, setResult] = useState<ProduceResult | undefined>();
-  const [error, setError] = useState<unknown | undefined>();
-
-  const isError = Boolean(error);
-  const isSuccess = !isUninitialized && !isLoading && !isError;
-
-  const requestState = {
-    isUninitialized,
-    isLoading,
-    isSuccess,
-    isError,
-    result,
-    error,
-  };
+  const [fetchRuleGroup, requestState] = alertRuleApi.endpoints.getRuleGroupForNamespace.useLazyQuery();
 
   /**
-   * This function will fetch the latest we have on the rule group, apply a diff to it via a reducer and sends
-   * the new rule group update to the correct endpoint.
+   * This function will fetch the latest configuration we have for the rule group, apply a diff to it via a reducer and sends
+   * returns the result.
    *
    * The API does not allow operations on a single rule and will always overwrite the existing rule group with the payload.
    *
-   * ┌─────────────────────────┐  ┌───────────────┐  ┌───────────────────┐
-   * │ fetch latest rule group │─▶│ apply reducer │─▶│ update rule group │
-   * └─────────────────────────┘  └───────────────┘  └───────────────────┘
+   * ┌─────────────────────────┐  ┌───────────────┐  ┌──────────────────┐
+   * │ fetch latest rule group │─▶│ apply reducer │─▶│  new rule group  │
+   * └─────────────────────────┘  └───────────────┘  └──────────────────┘
```

Comment:
I think it would be beneficial to place this file in a separate folder. Since we plan to use the same pattern for all entities, this will make it easier to locate and manage the available hooks. Wdyt?

Reply:
I do plan to make separate hooks for the other higher-order hooks so this one will just contain the logic for fetch + patch. I've opted not to do that in this PR as it would make it too big :D